core/LSU.hpp

```diff
@@ -26,29 +27,37 @@
 #include "FlushManager.hpp"
 #include "CacheFuncModel.hpp"
 #include "MemoryAccessInfo.hpp"
+#include "LoadStoreInstInfo.hpp"
 #include "MMU.hpp"
 #include "DCache.hpp"
 
 namespace olympia
 {
     class LSU : public sparta::Unit
     {
-    public:
+      public:
         /*!
          * \class LSUParameterSet
          * \brief Parameters for LSU model
          */
         class LSUParameterSet : public sparta::ParameterSet
         {
-        public:
+          public:
             //! Constructor for LSUParameterSet
-            LSUParameterSet(sparta::TreeNode* n):
-                sparta::ParameterSet(n)
-            {
-            }
+            LSUParameterSet(sparta::TreeNode* n) : sparta::ParameterSet(n) {}
 
             // Parameters for ldst_inst_queue
             PARAMETER(uint32_t, ldst_inst_queue_size, 8, "LSU ldst inst queue size")
+            PARAMETER(uint32_t, replay_buffer_size, ldst_inst_queue_size, "Replay buffer size")
+            PARAMETER(uint32_t, replay_issue_delay, 3, "Replay Issue delay")
+            // LSU microarchitecture parameters
+            PARAMETER(
+                bool, allow_speculative_load_exec, true,
```

Comment:
@h0lyalg0rithm is this parameter's boolean meaning inverted?   I'm looking at the code and if this is set to `true`, then the LS will NOT allow a load to bypass an older store.

Reply:
@klingaard This parameter by default is set to true, so it should allow a load to complete before a store